src/HOC/withAuth.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+import * as React from 'react';
+import { connect } from 'react-redux';
+import type { StoreType } from '../store/reducers';
+import type { State as SessionState } from '../store/reducers/SessionReducer';
+import { saveToken, signIn } from '../store/actions/SessionActions';
+import { getCookieToken } from '../helpers/Auth';
+
+type Props = {
+  doSignIn: Function,
+  doSaveToken: Function,
+};
+
+function withAuth(BaseComponent: React.ComponentType<any>) {
+  class WithAuth extends React.Component<Props> {
+    constructor(props) {
+      const JWT_Token = getCookieToken();
+      if (JWT_Token) {
+        props.doSaveToken(JWT_Token);
+      }
+      super(props);
```

Comment:
I would personally advice to call `super` as the first thing in constructor, although it's not necessary until you use `this`